pkg/types/nutanix/machinepool.go

```diff
@@ -0,0 +1,57 @@
+package nutanix
+
+// MachinePool stores the configuration for a machine pool installed
+// on Nutanix.
+type MachinePool struct {
+	// NumCPUs is the total number of virtual processor cores to assign a vm.
+	//
+	// +optional
+	NumCPUs int32 `json:"cpus"`
```

Comment:
Optionals should have `omitempty`.
```suggestion
	NumCPUs int32 `json:"cpus,omitempty"`
```